### DISCUSSION TITLE
How to get the default camera resolution for CameraView?

### DISCUSSION THREAD

**Discussion by codychaplin** [CommunityToolkit/Maui#2220] [Q&A]
By default, the `ImageCaptureResolution` is set to (0,0) (see screenshot below). How can I find out what it actually is without manually setting it? I am trying to find all the supported resolutions with the same aspect ratio as what the default camera preview shows because, when you change the resolution, it doesn't update the preview, meaning what I see in the preview is not exactly what will get captured in the image.
![Screenshot 2024-09-21 170451](https://github.com/user-attachments/assets/ad1e3ca1-b0db-457f-b4bc-87db2f61a9a3)

**Comment by bijington** [comment]
I think we should probably update the preview resolution to match the `ImageCaptureResolution` the issue comes with what to show when the aspect ratio doesn't match...

For now though have you tried looking at the `SupportedResolutions` property from the `CameraInfo` class?

**Reply by codychaplin** [reply]
Yeah my current implementation takes the ratio of the first one in that list and uses that to filter the rest of them. The only thing is, the resolution of the second one in the list is what my actual camera app on my phone (Pixel 7) takes by default. It's close enough to the first, but I'm just making the assumption that's the case for other devices for now.

**Reply by bijington** [reply]
It looks like on Apple platforms it takes the largest resolution by default. It certainly sounds like there is a bug if it isn't setting the ImageCaptureResolution property when it makes use of a default.
